simulariumio/data_objects/agent_data.py

```diff
@@ -536,9 +536,35 @@ def display_type_for_agent(self, time_index: int, agent_index: int) -> DISPLAY_T
         """
         type_name = self.types[time_index][agent_index]
         if type_name not in self.display_data:
-            raise DataError(f"{type_name} has no DisplayData, cannot get display_type")
+            self.display_data[type_name] = DisplayData(
+                name=type_name,
+                display_type=self._default_display_type_for_agent(
+                    time_index, agent_index
+                ),
+            )
         return self.display_data[type_name].display_type
 
+    def _default_display_type_for_agent(
+        self, 
+        time_index: int, 
+        agent_index: int
+    ) -> DISPLAY_TYPE:
+        """
+        Get the default DISPLAY_TYPE to use 
+        given the number of subpoints for the agent
+        at the given time and agent indices
+        """
+        n_subpoints = self.n_subpoints[time_index][agent_index]
+        default_display_types = [
+            DISPLAY_TYPE.FIBER, 
+            DISPLAY_TYPE.SPHERE_GROUP, 
+            DISPLAY_TYPE.SPHERE,
+        ]
+        for display_type in default_display_types:
+            if n_subpoints % float(SUBPOINT_VALUES_PER_ITEM(display_type)) == 0:
+                return display_type
```

Comment:
there is something funky about this indirection..  are you relying on the fact that the subpoint values per item is different for all three?

Reply:
yeah I am. I could make a constant dict or something with default to use for each values_per_item

Reply:
I agree with that, make it more explicit, or at least just a comment saying "if the number of points per type ever changes then we have to rethink this"